pkg/graphgen/union.go

```diff
@@ -0,0 +1,76 @@
+package graphgen
+
+import (
+	"sync"
+
+	"github.com/arangodb/feed/pkg/datagen"
+)
+
+type UnionParameters struct {
+	Left       Generatable
+	Right      Generatable
+	Prefix     string
+	StartIndex uint64
+}
+
+func (u UnionParameters) MakeGraphGenerator() (GraphGenerator, error) {
+	V := make(chan *datagen.Doc, batchSize())
+	E := make(chan *datagen.Doc, batchSize())
+
+	p1, errLeft := u.Left.MakeGraphGenerator()
+	if errLeft != nil {
+		return nil, errLeft
+	}
+	p2, errRight := u.Right.MakeGraphGenerator()
+	if errRight != nil {
+		return nil, errRight
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		for v := range p1.VertexChannel() {
+			V <- v
+		}
+		wg.Done()
+	}()
+
+	go func() {
+		for v := range p2.VertexChannel() {
+			V <- v
+		}
+		wg.Done()
+	}()
+
+	go func() {
```

Comment:
I think this combination is OK. But I am worried about the keys. I think before long we need to agree if the keys of the vertices are produced by the graph generator or by my generic graph importer. Here it seems that it is better if the graph generator produces the keys, otherwise we have no control in which order the combined channel of vertices arrives at my tool, and so we have no way to tell what the keys of individual vertices will be. Therefore, the edges cannot get their _from and _to right. So I fear we must move to that the graph generator produces the keys.